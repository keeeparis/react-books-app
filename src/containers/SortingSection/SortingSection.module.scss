.wrapper {
  display: flex;
  gap: 40px;
}

@media (max-width: 660px) {
  .wrapper {
    flex-direction: column;
    gap: 10px;
  }
}
