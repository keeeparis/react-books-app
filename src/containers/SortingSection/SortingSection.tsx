import React, { ChangeEvent } from 'react'
import Select from '../../components/Select'
import { categories, sorting } from '../../mock'
import { useAppDispatch, useAppSelector } from '../../redux/hooks/hooks'
import {
  updateCategory,
  updateSorting,
} from '../../redux/mainFeature/mainSlice'
import {
  selectCategory,
  selectSorting,
} from '../../redux/mainFeature/selectors'
import styles from './SortingSection.module.scss'

const SortingSection = () => {
  const dispatch = useAppDispatch()
  const category = useAppSelector(selectCategory)
  const sort = useAppSelector(selectSorting)

  const onCategoryChange = (option: string) => {
    dispatch(updateCategory(option))
  }

  const onSortingChange = (option: string) => {
    dispatch(updateSorting(option))
  }

  return (
    <div className={styles.wrapper}>
      <Select data={categories} onClick={onCategoryChange} value={category} />
      <Select data={sorting} onClick={onSortingChange} value={sort} />
    </div>
  )
}

export default SortingSection
