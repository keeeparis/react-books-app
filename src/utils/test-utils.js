import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
