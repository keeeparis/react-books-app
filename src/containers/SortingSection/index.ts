import SortingSection from './SortingSection'

export default SortingSection
