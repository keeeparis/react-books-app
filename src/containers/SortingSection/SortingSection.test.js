import { render, screen, fireEvent, userEvent } from 'test-utils'
import SortingSection from './SortingSection'

it('test SortingSection Component', () => {
  const { asFragment } = render(<SortingSection />)

  expect(asFragment()).toMatchSnapshot()
})
