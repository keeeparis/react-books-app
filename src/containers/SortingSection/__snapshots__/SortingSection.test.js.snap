// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SortingSection Component 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="select"
    >
      <div
        class="selectShow"
        role="selected"
        tabindex="0"
      >
        all
      </div>
    </div>
    <div
      class="select"
    >
      <div
        class="selectShow"
        role="selected"
        tabindex="0"
      >
        relevance
      </div>
    </div>
  </div>
</DocumentFragment>
`;
